### ISSUE TITLE
Defining target for multilabel/multiclass outputs 

### ISSUE THREAD

**Issue by kumudjk** [fchollet/keras#4737]
I am using Keras to predict multilables, that is, one record might have multiple classes. For ex, if there are 10 target classes, a record can have more than one target class. In this scenario whats the best way to transform target. I am using categorical_crossentropy as my objective. I tried both - 1) having multiple 1s and 0s within 10 output vectors & 2) having only 1 in one of the 10 output vectors & rest 0s. In this way I would duplicate each record with multiple outputs.
I read Keras link https://keras.io/objectives/ where its specifically written - 
"Note: when using the categorical_crossentropy objective, your targets should be in categorical format (e.g. if you have 10 classes, the target for each sample should be a 10-dimensional vector that is all-zeros expect for a 1 at the index corresponding to the class of the sample). In order to convert integer targets into categorical targets, you can use the Keras utility to_categorical"
If this is true, still I don't get any error with with 1st approach and even the predictions are better. I want to understand whats happening inside and how does it tackle this kind of situations

**Comment by keunwoochoi** [comment]
I don't think the second approach is valid. You're then giving a true signal and a false signal for the exactly same input. You can apply some sort of LDA to reduce the dimension of label spaces. If the label space is much smaller than 2**10 (which is the maximum number of cases), you can convert them into N (>10 and <2**10) different labels and then use categorical_crossentropy. Or use `sigmoid` as output activation and use `mse` or `binary_crossentropy` as a metric.

**Comment by kumudjk** [comment]
Hmm, I got your point. Logically it make sense!
I wanted to understand what happens when I feed 1st scenario in the model? How does prediction happens? According to the documentation, 2nd scenario should be the ideal or correct way of creating targets (forgetting its duplicates). Still I don't get any error with 1st scenario so curios to know what exactly is happening behind the scene
